### ISSUE TITLE
Transformation fails with not meaningful exception: Cannot read property 'id' of null

### ISSUE THREAD

**Issue by dunnock** [codemix/flow-runtime#49]
This is a:

 - [x] Bug Report

Which concerns:

 - [x] babel-plugin-flow-runtime

---

### What is the current behaviour?
`UploadProgressReporter.js:`
```
type TaskResult = {
  result: string,
  type: string
};

const BATCH_SIZE = 10;

class UploadProgressReporter {
  server: Hapi$Server;
  Jobs: App$Model$Job;
  job: Model$Job;
  mq: Hapi$Queue;

  constructor(job: Model$Job, req: Hapi$Request) {
    this.server = req.server;
    this.Jobs = req.server.plugins['hapi-mongo-models'].Jobs;
    this.mq = req.server.plugins.queue;
    this.job = job;
  }

  async monitor(tasks: Array<Promise<TaskResult>>): Promise<Array<string>> {
    const _id = this.job._id;
    if(!_id) return ["no job id"];
    let ret = [];

    for(let i=0; i<tasks.length; i+=BATCH_SIZE) {
      const promises = tasks.slice(i, i+BATCH_SIZE);
      try {
        const results = await Promise.all(promises);
        await this.publish(results);
        ret.concat(results.map(({result})=>result));
      } catch(e) {
        this.job.errors.push(e.message);
      }
    }

    try {
      console.log("replace job", await this.Jobs.replaceOne({_id}, this.job));
    } catch(e) {
      this.server.log(['error', 'mongo', 'UploadProgressReporter'], e.message);
    }

    return ret;
  }

  async publish(results: Array<TaskResult>) {
    const ret = results.forEach(({type}) => {
        this.job[type].processed = this.job[type].processed + 1;});
    this.job.updated = new Date();
    await this.mq.publish('events', this.job.userId.toString(), this.job, false);
    return ret;
  }

};

async function task(promise: Promise<string>, type: string): Promise<TaskResult> {
  try {
    const result = await promise;
    return { type, result }
  } catch(err) {
    return { type, result: err.message }
  }
}

export { task };

export default UploadProgressReporter;
```

`Error while processing:`
```
TypeError: server/api/snapshots/UploadProgressReporter.js: Cannot read property 'id' of null
    at ConversionContext.hasForwardTypeDeclaration (/node_modules/babel-plugin-flow-runtime/lib/ConversionContext.js:266:38)
    at ConversionContext.hasTDZIssue (/node_modules/babel-plugin-flow-runtime/lib/ConversionContext.js:238:21)
    at Identifier (node_modules/babel-plugin-flow-runtime/lib/firstPassVisitors.js:56:21)
    at NodePath._call (/node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (/node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (/node_modules/babel-traverse/lib/path/context.js:105:12)
    at TraversalContext.visitQueue (/node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitSingle (/node_modules/babel-traverse/lib/context.js:108:19)
    at TraversalContext.visit (/node_modules/babel-traverse/lib/context.js:192:19)
    at Function.traverse.node (/node_modules/babel-traverse/lib/index.js:114:17)
```

Please note, I load types via libs section of `.flowconfig`

---

### What is the expected behaviour?
Transformation succeeds or raises meaningful error. Please note, flow checks succeed on the same file.

---

### Which package versions are you using?
flow-runtime an babel-plugin-flow-runtime 0.3.0

**Comment by bluepnume** [comment]
Thanks for the fix! Really nice tool by the way, works great!

**Comment by phpnode** [comment]
fixed in 0.4.0
